pkg/sql/create_table.go

```diff
@@ -1210,7 +1210,7 @@ func MakeTableDesc(
 				if st == nil {
 					return desc, invalidClusterForShardedIndexError
 				}
-				if version := cluster.Version.ActiveVersionOrEmpty(ctx, st); version == (clusterversion.ClusterVersion{}) ||
+				if version := st.Version.ActiveVersionOrEmpty(ctx); version == (clusterversion.ClusterVersion{}) ||
```

Comment:
i think this should be !=

Reply:
Meant to throw a `!` in there, but that aside, I do think we want to error out if the version is not initialized, no?

Reply:
(I don't really know what's going on with `version != (clusterversion.ClusterVersion{})` above, we're not really handling the uninitialized version properly I think.)

Reply:
I'm going to take a look at this soon (#45519), so i wouldn't sweat about this. When i can move it to `ActiveVersion` it will error our if the version is not initialized.